pkg/inject/sidecar.go

```diff
@@ -41,8 +41,13 @@ const (
 func Sidecar(jaeger *v1.Jaeger, dep *appsv1.Deployment) *appsv1.Deployment {
 	deployment.NewAgent(jaeger) // we need some initialization from that, but we don't actually need the agent's instance here
 	logFields := jaeger.Logger().WithField("deployment", dep.Name)
-
-	if jaeger == nil || (dep.Annotations[Annotation] != jaeger.Name && dep.Annotations[AnnotationLegacy] != jaeger.Name) {
+	// Three conditions for inject:
+	// 	- jaeger not nil
+	// 	- jaeger instance name equal to annotation or annotation true (including legacy annotation).
+	//  - was not already injected (does not contain label)
+	labelValue, hasLabel := dep.Labels[Label]
+	hasInvalidAnnotation := dep.Annotations[Annotation] != jaeger.Name && dep.Annotations[Annotation] != "true" && dep.Annotations[AnnotationLegacy] != jaeger.Name
+	if jaeger == nil || hasInvalidAnnotation || (hasLabel && labelValue != jaeger.Name) {
```

Comment:
`(hasLabel && labelValue != jaeger.Name)` perhaps it can be `labelValue != jaeger.Name`

Reply:
not the same, the first time this function is called (and the deployment does not have an injected side-car), the label does not exist, in successive calls it should exist, but is only re-inject the side-car if is the same Jaeger instance. May be the second part is not entirely necessary.